src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js

```diff
@@ -159,10 +161,16 @@ export class RuleEditor extends LinkedComponent {
             },
             error => this.setState({ error, isPending: false, changesApplied: true })
           );
+        logEvent(this.ruleToDiagnosticsModel(requestProps, 'RuleApplyClick'));
       }
     }
   }
 
+  ruleToDiagnosticsModel = (eventName, rule) => {
```

Comment:
Why not make this whole function move to models?

Or... 
```js
 ruleToDiagnosticsModel = (eventName, rule) => toDiagnosticsModel(eventName, toLogRuleModel(rule));
```

Reply:
done